targets/TARGET_STM/trng_api.c

```diff
@@ -51,14 +51,18 @@ void trng_init(trng_t *obj)
 #if defined(DUAL_CORE)
     while (LL_HSEM_1StepLock(HSEM, CFG_HW_RCC_SEMID)) {
     }
+
+    LL_HSEM_ReleaseLock(HSEM, CFG_HW_RCC_SEMID, HSEM_CR_COREID_CURRENT);
 #endif /* DUAL_CORE */
+
+#elif defined(TARGET_STM32WL)
+    RCC_PeriphCLKInitTypeDef PeriphClkInitStruct;
+    /*Select PLLQ output as RNG clock source*/
+    PeriphClkInitStruct.PeriphClockSelection = RCC_PERIPHCLK_RNG;
+    PeriphClkInitStruct.RngClockSelection = RCC_RNGCLKSOURCE_MSI;
     if (HAL_RCCEx_PeriphCLKConfig(&PeriphClkInitStruct) != HAL_OK) {
         error("RNG clock configuration error\n");
     }
-#if defined(DUAL_CORE)
```

Comment:
You removed H7 code...

Reply:
Yes because we have this line before
if defined(TARGET_STM32H7)

Reply:
Done